test/server/graphql/v2/mutation/MemberInvitationMutations.test.ts

```diff
@@ -0,0 +1,437 @@
+      $since: DateTime
+    ) {
+      inviteMember(
+        memberAccount: $memberAccount
+        account: $account
+        role: $role
+        description: $description
+        since: $since
+      ) {
+        id
+        role
+        description
+        since
+      }
+    }
+  `;
+
+  describe('inviteMember', () => {
+    it('should create a new member invitation and its related activity', async () => {
+      sendEmailSpy.resetHistory();
+      const randomUserToInvite = await fakeUser();
+      const result = await utils.graphqlQueryV2(
+        inviteMemberMutation,
+        {
+          memberAccount: { id: idEncode(randomUserToInvite.collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          account: { id: idEncode(collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          description: 'new user 2 as MEMBER',
+          role: roles.MEMBER,
+          since: new Date('01 January 2022').toISOString(),
+        },
+        collectiveAdminUser,
+      );
+
+      const activity = await models.Activity.findAll({
+        where: {
+          type: ActivityTypes.COLLECTIVE_CORE_MEMBER_INVITED,
+          CollectiveId: collective.id,
+        },
+      });
+      const createdMemberInvitation = result.data.inviteMember;
+
+      expect(result.errors).to.not.exist;
+      expect(activity.length).to.equal(1);
+      expect(activity[0].data.memberCollective.id).to.equal(randomUserToInvite.collective.id);
+      expect(createdMemberInvitation.role).to.equal(roles.MEMBER);
+      expect(createdMemberInvitation.description).to.equal('new user 2 as MEMBER');
+      expect(createdMemberInvitation.since.toISOString()).to.equal(new Date('01 January 2022').toISOString());
+      expect(sendEmailSpy.callCount).to.equal(1);
+      expect(sendEmailSpy.args[0][1]).to.equal('Invitation to join webpack test collective on Open Collective');
+    });
+    it('must be authenticated as an admin of the collective', async () => {
+      sendEmailSpy.resetHistory();
+      const randomUser = await fakeUser();
+      const anotherRandomUser = await fakeUser();
+      const result = await utils.graphqlQueryV2(
+        inviteMemberMutation,
+        {
+          memberAccount: { id: idEncode(anotherRandomUser.id, IDENTIFIER_TYPES.ACCOUNT) },
+          account: { id: idEncode(collective.id, IDENTIFIER_TYPES.ACCOUNT) },
+          description: 'new user 2 as MEMBER',
+          role: roles.MEMBER,
+          since: new Date('01 January 2022').toISOString(),
+        },
+        randomUser,
+      );
+      expect(sendEmailSpy.callCount).to.equal(0);
+      expect(result.errors).to.have.length(1);
+      expect(result.errors[0].message).to.equal('Only admins can send an invitation.');
+    });
+    it('can only add with role accountant, admin, or member', async () => {
+      // Able to add ACCOUNTANT Role
+      sendEmailSpy.resetHistory();
+      const randomUserToInviteAsAccountant = await fakeUser();
+      const result1 = await utils.graphqlQueryV2(
```

Comment:
:ice_cream: This test could have been shorter and simpler by doing something like:

```es6
[roles.ACCOUNTANT, roles.ADMIN, roles.MEMBER].forEach(role => {
  // Tests for role here
})
```